### PR TITLE
Fix/allow to discard jwt by expiration

### PR DESCRIPTION
**Pull request by labajo**

### CODE REVIEW

File: lib/jwt-utils.js [hunk 1]
```diff
@@ -364,6 +364,14 @@ function checkPayload(token, cb) {
       return cb(errors.INVALID_EXP());
     }
 
+    var clientExpired = iatParsed + config.expiration;
```
Comment (dlozlla) [inline lib/jwt-utils.js]: I don't understand the name "clientExpired"... what about expLimit? We should also align the change in the name to the errors.

Also, what happens if there is no iat in the incoming token?

File: lib/jwt-utils.js [hunk 2]
```diff
@@ -364,6 +364,14 @@ function checkPayload(token, cb) {
...
+
+    if (clientExpired < payload.exp) {
```
Comment (dlozlla) [inline lib/jwt-utils.js]: This condition is not really required. You can directly compare clientExpired with currentDate and the palyload.exp with currentDate, or select the more restrictive expiration and the compare with current date.

File: lib/jwt-utils.js [hunk 3]
```diff
@@ -364,6 +364,14 @@ function checkPayload(token, cb) {
...
+      if (clientExpired < currentDate) {
+        return cb(errors.EXPIRED_JWT_BY_CLIENT(), token);
+      }
+    }
+
```
Comment (dlozlla) [inline lib/jwt-utils.js:369]: All this is included in an if (payload.exp)... what if the incoming jwt does not have exp? we should calculate the expLimit in that case too.

File: lib/jwt-utils.js (hunk 1 above)
Comment (labajo) [inline lib/jwt-utils.js]: return in the callback errors.INVALID_IAT()

**Comment by dlozlla** [comment]
LGTM

**Pull request merged**